eox_nelp/signals/receivers.py

```diff
@@ -36,6 +37,29 @@ def block_completion_progress_publisher(instance, **kwargs):  # pylint: disable=
     )
 
 
+def emit_initialized_course_event(instance, **kwargs):  # pylint: disable=unused-argument
+    """This receiver is connected to the post_save BlockCompletion signal
+    and this checks how many BlockCompletion records exists, if there is only one record
+    the `nelc.eox_nelp.initialized.course` event will be emitted.
+
+    Args:
+        instance<Blockcompletion>: Instance of BlockCompletion model.
+    """
+    completion_blocks = instance.user_learning_context_completion_queryset(instance.user, instance.context_key)
+
+    if len(completion_blocks) == 1:
```

Comment:
There is only for initalized and not for finish course. 
That the reason the qs is only 1 when there is only one in the completion queryset per user and course.?
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/9dee7c66-4b36-4b8c-9aa2-889043aff6ad)

Reply:
> There is only for initalized and not for finish course event? That the reason the qs is only 1 when there is only one in the completion queryset per user and course.? ![image](https://private-user-images.githubusercontent.com/51926076/289695880-9dee7c66-4b36-4b8c-9aa2-889043aff6ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzMzYyMzgsIm5iZiI6MTcwMjMzNTkzOCwicGF0aCI6Ii81MTkyNjA3Ni8yODk2OTU4ODAtOWRlZTdjNjYtNGIzNi00YjhjLTlhYTItODg5MDQzYWZmNmFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDIzMDUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhYzRiZGRiMmIwZTNhNzBlNmRlY2U3MTZjOTQ3MzAxM2ZkMGEyMTYzMGFlYWFiZDM5MDgwMzhiNGExYTM0NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.b9O32m2qioRsAAfeT2JbOA00a8rpOQaz64DP1RSfdEc)

The finish event is the course completion event that is generated by aggregators  and I didn't catch your second question